### PR TITLE
Extended typings based on pugjs.org doc.

### LINKED ISSUE #1: Missing some function declarations
There no declarations of some pug functions like `compileClient` or `compileClientWithDependenciesTracked`. The old jade typings are compatible?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cc3d223a946f661eff871787edeb0fcb8f0db156/jade/jade.d.ts
Could you extend this typings to match with pug API?

### PR DESCRIPTION
**Pull request by MichalLytek**
Fixed #1.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,19 +1,79 @@
...
+    /**
+     * Compile a Pug template to an object of the form:
+     * {
+     *   'body': 'function (locals) {...}',
+     *   'dependencies': ['filename.pug']
+     * }
+     * that can be used client side along with the Pug runtime.
+     * You should only use this method if you need dependencies to implement something like watching for changes to the Pug files.
+     */
+    export function compileClientWithDependenciesTracked(template: string, options?: any): {
```
Comment (blakeembrey) [inline index.d.ts]: Is this `options` here the same or different to the rest?

**Comment by blakeembrey** [comment]
The only other thing is stylistic - where possible, it's nice to follow the existing style guidelines set up. In this case, you refactored it all so no big deal on the code style. Indent-wise, it should be 2 (https://github.com/types/npm-pug/blob/master/.editorconfig#L6).

**Comment by MichalLytek** [comment]
@blakeembrey, thanks for info, I'm new here :wink: Changed it in 649267e.

**Comment by blakeembrey** [comment]
Looks fantastic! 😄

File: index.d.ts (hunk 1 above)
Comment (MichalLytek) [reply index.d.ts]: It's the same, I forgot to change it.
Checked in https://github.com/pugjs/pug/blob/master/lib/index.js.

**Pull request merged**